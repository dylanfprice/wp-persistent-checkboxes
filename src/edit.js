import { RichText } from '@wordpress/block-editor';

import parseLabels from './parseLabels';

export default function	edit( { attributes: { labelObjects }, className, setAttributes } ) {
	const content = (
		labelObjects.map( ( { label } ) => `<p>${ label }</p>` ).join( '' )
	);
	const style = `
      .${ className } p:before {content: "\u2610 ";}
      .${ className } p {margin-bottom: 0 !important;}
    `;
	return (
		<div className={ className }>
			<style>{ style }</style>
			<RichText
				multiline="p"
				value={ content }
				onChange={ ( content ) => setAttributes( {
					labelObjects: parseLabels( content ).map( ( label ) => {
						return { label };
					} ),
				} ) }
			/>
		</div>
	);
}
