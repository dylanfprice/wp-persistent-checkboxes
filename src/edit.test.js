import { RichText } from '@wordpress/block-editor';
import { shallow } from 'enzyme';

import edit from './edit';

describe( 'edit', () => {
	it( 'renders the RichText editor', () => {
		const labelObjects = [
			{ label: 'label one' },
			{ label: 'label two' },
			{ label: 'label three' },
		];
		const editComponent = shallow( edit( {
			attributes: { labelObjects },
			className: 'test',
		} ) );
		expect( editComponent ).toMatchSnapshot();
	} );

	it( 'sets content from label objects', () => {
		const editComponent = shallow( edit( {
			attributes: { labelObjects: [ { label: 'test label' } ] },
			className: 'test',
		} ) );
		const actual = editComponent.find( RichText ).prop( 'value' );
		expect( actual ).toBe( '<p>test label</p>' );
	} );

	it( 'updates label objects from content when content changes', () => {
		const setAttributes = jest.fn();
		const editComponent = shallow( edit( {
			attributes: { labelObjects: [] },
			className: 'test',
			setAttributes,
		} ) );
		editComponent.find( RichText ).simulate( 'change', '<p>test label</p>' );
		const expected = [
			[ { labelObjects: [ { label: 'test label' } ] } ],
		];
		expect( setAttributes.mock.calls ).toEqual( expected );
	} );
} );
