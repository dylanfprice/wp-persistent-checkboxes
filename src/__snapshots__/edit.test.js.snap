// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`edit renders the RichText editor 1`] = `
<div
  className="test"
>
  <style>
    
      .test p:before {content: "‚òê ";}
      .test p {margin-bottom: 0 !important;}
    
  </style>
  <WithInstanceId(WithBlockEditContext(WithSelect(WithDispatch(RichTextWrapper))))
    multiline="p"
    onChange={[Function]}
    value="<p>label one</p><p>label two</p><p>label three</p>"
  />
</div>
`;
